### ISSUE TITLE
Adan Optimizer

### ISSUE THREAD

**Issue by joaogui1** [google-deepmind/optax#401]
Is there any interest in adding the [Adan optimizer](https://twitter.com/davisblalock/status/1561976182567870465?t=AHwO3of5ivzgE0dW06hZvA&s=35) to optax? If so I can do it

**Comment by mkunesch** [comment]
Interesting, I think that would be great! Thanks a lot! 

Let us know if you'd like to discuss anything about the implementation as you write it. How much of the Adam code do you think can be reused?

**Comment by mtthss** [comment]
This looks very interesting, it would be an amazing contribution!

**Comment by adam-hartshorne** [comment]
I see there is an issue with the replicability of the pull request. It appears there is another implementation in jax for optax here, which might be worth looking at.

https://github.com/hr0nix/optax-adan

**Comment by joaogui1** [comment]
Thanks for the pointer @adam-hartshorne! Sadly when testing that implementation on my colab it gets a 3 order of magnitude larger error compared to mine (not sure where the difference comes from, the only difference I can see is the epsilon placement)

**Comment by hr0nix** [comment]
Hi, author of optax-adan here. Is it possible to share a collab where you've compared both implementations? I'd like to figure out where does the difference come from.

**Comment by hr0nix** [comment]
Oh, found the link in the pull request, so no worries.

**Comment by hr0nix** [comment]
Yep, the difference between implementations comes from epsilon placement. If I move it outside sqrt, the results are equal.

**Comment by hr0nix** [comment]
After this change is merged and released, I'll put a note in README.md of optax-adan that there is no need to use the package as adan is now implemented in optax.

**Comment by vroulet** [comment]
Closed by #1090
